lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2007/179d_ashrae_90_1_2007/179d_ashrae_90_1_2007.SubSurface.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class ACM179dASHRAE9012007
+
+  def get_exterior_fenestration_value(sub_surface, column_name)
+    known_columns = [
+      'Construction',
+      'Frame and Divider',
+      'Glass Area',
+      'Frame Area',
+      'Divider Area',
+      'Area of One Opening',
+      'Area of Multiplied Openings',
+      'Glass U-Factor',
+      'Glass SHGC',
+      'Glass Visible Transmittance',
+      'Frame Conductance',
+      'Divider Conductance',
+      'NFRC Product Type',
+      'Assembly U-Factor',
+      'Assembly SHGC',
+      'Assembly Visible Transmittance',
+      'Shade Control',
+      'Parent Surface',
+      'Azimuth',
+      'Tilt',
+      'Cardinal Direction',
+    ]
+    raise "Unknown column '#{column_name}'. Available: #{known_columns}" unless known_columns.include?(column_name)
+
+    sql_query = """
+SELECT Value FROM TabularDataWithStrings
+  WHERE ReportName='EnvelopeSummary'
+    AND ReportForString='Entire Facility'
+    AND TableName='Exterior Fenestration'
+    AND RowName='#{sub_surface.nameString().upcase}'
+    AND ColumnName='#{column_name}'
+"""
+
+    val_ = sub_surface.model.sqlFile.get.execAndReturnFirstDouble(sql_query)
+    raise "Query failed: #{sql_query}" if val_.empty?
+    return val_.get
+  end
```

Comment:
Helper to query the Exterior Fenestration table

Basically reimplement but slightly differently the https://github.com/NREL/OpenStudio/blob/d8eac889d0fb8f16c42fcedc3bd92217a98af4f6/src/model/SubSurface.cpp#L1318